akka-actor/src/main/scala/akka/pattern/internal/BackoffOnRestartSupervisor.scala

```diff
@@ -94,9 +94,10 @@ import akka.pattern.{ BackoffReset, BackoffSupervisor, HandleBackoff }
     case Some(c) =>
       c.forward(msg)
     case None =>
-      replyWhileStopped match {
-        case None    => context.system.deadLetters.forward(msg)
-        case Some(r) => sender() ! r
+      handlerWhileStopped match {
+        case Some(Left(r))  => sender() ! r
+        case Some(Right(h)) => h.forward(msg)
+        case None           => context.system.deadLetters.forward(msg)
```

Comment:
Not super important since internal, but perhaps encode this in an internal sealed ADT instead of nesting Option and Either, given that it basically is three possible states now?

Reply:
Yep, pushed an update with a sealed ADT, makes it more readable 👍